x/cork/keeper/msg_server.go

```diff
@@ -62,3 +62,37 @@ func (k Keeper) SubmitCork(c context.Context, msg *types.MsgSubmitCorkRequest) (
 
 	return &types.MsgSubmitCorkResponse{}, nil
 }
+
+// ScheduleCork
+func (k Keeper) ScheduleCork(c context.Context, msg *types.MsgScheduleCorkRequest) (*types.MsgScheduleCorkResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	if !k.HasCellarID(ctx, common.HexToAddress(msg.Cork.TargetContractAddress)) {
+		return nil, types.ErrUnmanagedCellarAddress
+	}
+
+	signer := msg.MustGetSigner()
+	validatorAddr, err := k.signerToValAddr(ctx, signer)
+	if err != nil {
+		return nil, err
+	}
+
+	k.SetCork(ctx, validatorAddr, *msg.Cork)
```

Comment:
Shouldn't we be using `SetScheduledCork`?

Reply:
Yes. Corrected